src/services/pending-queue.ts

```diff
@@ -169,9 +172,44 @@ export async function deletePendingQueueSubmission(pendingTimestamp: string) {
   await removePendingQueueSubmission(pendingTimestamp, 'DELETION')
 }
 
+/**
+ * Edit a PendingFormSubmission before it is processed based on the
+ * `pendingTimestamp` property. The function removes the submission from the
+ * pending queue and returns a prefill Id
+ *
+ * ### Example
+ *
+ * ```js
+ * const pendingTimestamp = '2020-07-29T01:03:26.573Z'
+ * const { preFillFormDataId, formId } =
+ *   await submissionService.editPendingQueueSubmission(pendingTimestamp)
+ * window.location.href = `https://mycoolforms.apps.oneblink.io/forms/${formId}?preFillFormDataId=${preFillFormDataId}`
+ * ```
+ *
+ * @param pendingTimestamp
+ */
+export async function editPendingQueueSubmission(
+  pendingTimestamp: string,
+): Promise<{ preFillFormDataId: string; formId: number }> {
+  try {
+    const formSubmission = await getFormSubmission(pendingTimestamp)
+    if (!formSubmission) {
+      throw new Error('Could not find formSubmision to edit')
+    }
+    await removePendingQueueSubmission(pendingTimestamp, 'EDIT')
+    const preFillFormDataId = uuidv4()
```

Comment:
```suggestion
    const preFillFormDataId = `PENDING_SUBMISSION_${pendingTimestamp}`
```

This will allow autosave data to be continued if they try to edit the same submission after navigating back to the pending queue screen.